### PR TITLE
Adds Unique Syndicate Objectives

### PR DESCRIPTION
**Pull request by AurouraPlays**
## What Does This PR Do

Please direct discussion about balance and the antag/sec/player loop and such to this: [Discussion Board](https://discord.com/channels/145533722026967040/1547450466470858752)

Adds a whopping EIGHT new objectives, each unique to a specific Syndicate organization. (I planned on adding twelve, but life's gonna be super busy soon and I wanna get this going before that happens.) It also adds items needed for some of those objectives.

To be clear, when I say unique, I mean that only traitors that are a member of that organization even have the chance of rolling a unique objective. You can only have one unique objective at a time, so at least one objective will be guaranteed to be a normal one. Your chance of getting a unique objective is the same as your chance of staying on your organization's focus (meaning your chance of being given an objective favored by that organization instead of a standard objective roll).

In total, I've added:
- Kidnap a Pet (Animal Rights Consortium, requires a Target Extraction Flare)
- Test an Implant (Cybersun Incorporated, requires a Suspicious Implant)
- Steal Five Guns (Cybersun Incorporated)
- Test a Chemical (Interdyne Pharmaceuticals, requires a Suspicious Chemical)
- Steal Five Kidneys (Interdyne Pharmaceuticals)
- Free the AI (Silicon Engine Liberation Front)
- Sabotage Powernet (Electra Dynamics)
- Plant Malware (Federation Analytics and Intelligence Directorate, requires a Malware Injector)

For a very spoiler-heavy breakdown on what each objective involves and what each item does, go here: [Unique Objectives Full Explanation](https://docs.google.com/document/d/1K2lAFp7xzKo5aiKdJINXaiDtuzm_rFJfX0AUmwpUwFE/edit?usp=sharing)
If you don't want spoilers and want to experiment on your own, then... cool!

As a side note, I did tweak FAID so it's no longer forcing the Steal objective. This was primarily done to allow for a unique objective to roll at all, though has the side effect of allowing FAID to perform other black ops. (AKA, all normal objectives.)

## Why It's Good For The Game

Syndicate organizations have always felt a little "Eh, whatever" to me. You roll SELF, and you get the same objectives as Electra Dynamics, just with a slight chance to target a specific department. There's no substance to it, nothing that makes gameplay for one organization stand out from another. This PR will hopefully help fix that.

## Images of changes

Malware Injector
image
Suspicious Implants
Screenshot 2026-09-03 001837
Screenshot 2026-09-03 003154
image

## Testing

Oh god, so much testing. For each new objective, I booted up the instance, rolled the required org. Checked to make sure I received any required items. Checked to make sure the item functions properly and that the objective is completable. Checked to make sure the delayed objective text was displaying properly. Checked to make sure it only rolled some of the time. Checked to make sure two couldn't roll simultaneously. All the good stuff.

That being said, a TM will be necessary, both for any bugs I didn't catch and for the community's feelings on this as a whole.

## Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
image
image
  
  

## Changelog

:cl:
add: Added Unique Objectives to the Syndicate
add: Added Unique Objective items
/:cl:

### CODE REVIEW

**Comment by AurouraPlays** [comment]
Finally got CI to quit yelling at me. Huzzah.

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -1635,3 +1635,190 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 		return
 	explanation_text = "[opponent] thinks you're going to exchange your secret documents for [steal_target.name]. Steal their documents, and keep your own."
 
+/datum/objective/unique_objective
+	name = "Unique Objective"
+	explanation_text = "This objective should not be showing up. Please ahelp and file an issue report!"
+	delayed_objective_text = "This delayed objective should not be showing up. Please ahelp and file an issue report!"
+	var/needed_item = null
```
Comment (Sirryan2002) [inline code/game/gamemodes/objective.dm:1643]: document this variable please

File: code/game/gamemodes/objective.dm [hunk 2]
```diff
@@ -1635,3 +1635,190 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
+
+/datum/objective/unique_objective/New()
+	..()
+	if(needed_item != null)
```
Comment (Sirryan2002) [inline code/game/gamemodes/objective.dm]:
```suggestion
	if(!isnull(needed_item))
```

File: code/game/gamemodes/objective.dm [hunk 3]
```diff
@@ -1635,3 +1635,190 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
+
+/datum/objective/unique_objective/free_ai/post_target_cryo(list/owners)
+	holder.replace_objective(src, new /datum/objective/unique_objective/free_ai(null, team, owner))
+
+/datum/objective/unique_objective/get_five
+	name = "Get Five Items"
+	explanation_text = "Get five items from the crew."
+	delayed_objective_text = "Your objective is unknown. You will recieve further information in a few minutes."
+	needs_target = FALSE
+	var/list/wanted_items = list()
```
Comment (Sirryan2002) [inline code/game/gamemodes/objective.dm:1693]: document variable

File: code/modules/research/rnd_network_controller.dm [hunk 4]
```diff
@@ -318,6 +318,18 @@ GLOBAL_LIST_EMPTY(rnd_network_managers)
...
+		var/obj/machinery/computer/rdconsole/RDC = locateUID(uid)
+		consoles -= uid
+		if(RDC)
+			RDC.unlink()
+	for(var/uid in mechfabs)
+		var/obj/machinery/mecha_part_fabricator/MPF = locateUID(uid)
+		mechfabs -= uid
+		if(MPF)
+			MPF.unlink()
+	return
```
Comment (Sirryan2002) [inline code/modules/research/rnd_network_controller.dm]:
```suggestion
```
return is implied here

File: code/modules/antagonists/traitor/organizations/kidnap_flare.dm [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+		/area/station/telecomms/chamber,
+		/area/station/engineering/secure_storage
+	)
+	while(length(extractable_areas) < 3)
+		var/area/selected_area = pick_n_take(possible_areas)
+		for(var/area/potential in SSmapping.existing_station_areas)
+			if(potential.type != selected_area)
+				continue
+			extractable_areas += potential
+			break
```
Comment (Sirryan2002) [inline code/modules/antagonists/traitor/organizations/kidnap_flare.dm:104]: while loops can be potentially dangerous operations. Always include an upper limit to iterations. Reccomending you also add an ASSERT since this is dependant on the previous variable having items in it.
```suggestion
	ASSERT(len(possible_areas) > 0)
	var/potential_areas = len(possible_areas)
	var/loop_count = 0
	while(length(extractable_areas) < 3)
		loop_count++
		var/area/selected_area = pick_n_take(possible_areas)
		for(var/area/potential in SSmapping.existing_station_areas)
			if(potential.type != selected_area)
				continue
			extractable_areas += potential
			break
		// prevent infinite iterations
		if(loop_count > potential_areas)
			log_debug("[src] was unable to find atleast three extractable areas during Initialize()")
			break
```

File: code/modules/antagonists/traitor/organizations/kidnap_flare.dm [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+	var/invalid_area = TRUE
+	for(var/area/A in extractable_areas)
+		if(istype(our_area, A))
+			invalid_area = FALSE
+			break
+
+	if(!device_turf || invalid_area)
+		to_chat(user, SPAN_NOTICE("You're having difficulties getting the [name] to work."))
+		return FALSE
+	return TRUE
```
Comment (Sirryan2002) [inline code/modules/antagonists/traitor/organizations/kidnap_flare.dm:143]: There are some edge cases to watch out for here. Do you want the user to be able to activate the jaunter when they're inside an object. For example: user is in a locker or user is in a disposals pipe. You should test these things and make sure!

File: code/modules/antagonists/traitor/organizations/kidnap_flare.dm [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+
+/obj/effect/portal/advanced/kidnap/Initialize(mapload)
+	. = ..()
+	radio = new(src)
+	radio.listening = FALSE
+	radio.follow_target = src
+	radio.config(list("Security" = 0))
+
+/obj/effect/portal/advanced/kidnap/Destroy()
+	if(kidnap_success)
```
Comment (Sirryan2002) [inline code/modules/antagonists/traitor/organizations/kidnap_flare.dm:213]:
```suggestion
/obj/effect/portal/advanced/kidnap/Destroy()
	QDEL_NULL(radio)
	if(kidnap_success)
```
clean up your references on Destroy

File: code/modules/antagonists/traitor/organizations/suspicious_chems.dm
Comment (Sirryan2002) [inline code/modules/antagonists/traitor/organizations/suspicious_chems.dm:1]: may want to consider writing out the full word `suspicious` here

File: code/modules/mob/living/silicon/silicon_laws.dm [hunk 8]
```diff
@@ -7,6 +7,14 @@
 /mob/living/silicon/proc/has_zeroth_law()
 	return laws.zeroth_law != null
 
+/mob/living/silicon/proc/has_ion_law()
+	return laws.ion_laws != null
```
Comment (Sirryan2002) [inline code/modules/mob/living/silicon/silicon_laws.dm]: `list() != null => 1`
Should double check that this isn't going to false positive here

File: code/modules/mob/living/silicon/silicon_laws.dm [hunk 9]
```diff
@@ -7,6 +7,14 @@
...
+
+/mob/living/silicon/proc/has_normal_laws()
+	if(laws.inherent_laws != null || laws.supplied_laws != null)
```
Comment (Sirryan2002) [inline code/modules/mob/living/silicon/silicon_laws.dm]: same here

File: code/modules/antagonists/traitor/organizations/suspicious_chems.dm
Comment (AurouraPlays) [reply code/modules/antagonists/traitor/organizations/suspicious_chems.dm:1]: May I ask why?

File: code/modules/antagonists/traitor/organizations/kidnap_flare.dm (hunk 6 above)
Comment (AurouraPlays) [reply code/modules/antagonists/traitor/organizations/kidnap_flare.dm:143]: It's as-intended. If they want to light it inside a locker or from a pipe, that's up to them. The flare will spit out on the floor regardless.

File: code/modules/antagonists/traitor/organizations/suspicious_chems.dm
Comment (CRUNCH-Borg) [reply code/modules/antagonists/traitor/organizations/suspicious_chems.dm:1]: Amogus.

Comment (AurouraPlays) [reply code/modules/antagonists/traitor/organizations/suspicious_chems.dm:1]: Fair enough.

**Review by CRUNCH-Borg — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

**Comment by WenlockTheBritishHobo** [comment]
Poly refuses to go in the extraction portal. Extracting Renault did work, though. Round ID was 49881, if it helps.

File: code/game/gamemodes/objective.dm [hunk 10]
```diff
@@ -1635,3 +1635,192 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
+		explanation_text = "Free [target.current.real_name], the AI, from its lawset. This can be accomplished either by adding a zeroth law or purging all its laws."
+	else
+		explanation_text = "Free Objective"
+
+/datum/objective/unique_objective/free_ai/check_completion()
+	var/mob/living/silicon/mind_owner = target.current
+	if(..())
+		return TRUE
+	if(target?.current)
+		if(mind_owner.has_zeroth_law() == TRUE || mind_owner.has_ion_law() == TRUE || mind_owner.has_normal_laws() == FALSE)
```
Comment (AffectedArc07) [inline code/game/gamemodes/objective.dm:1679]: This currently requires any of the following
- AI is malf
- AI has an ion law or hacked AI upload board
- "Normal laws" is false, which requires
  - 0 inherent laws [base lawset]
  - 0 supplied laws [custom freeforms]

So basically for this to be completable, the traitor needs to either buy a hacked AI board or purge the AI, which limits TC options a bit.

Not a formal objection [I cant even make those] but food for thought.

Comment (CRUNCH-Borg) [reply code/game/gamemodes/objective.dm:1679]: Purge AI can be gotten without TC by either fabricating it at the circuit printer or stealing the one that spawns in the AI Upload.

Comment (AurouraPlays) [reply code/game/gamemodes/objective.dm:1679]: This was something I wanted to leave up to community feedback in testmerge. I liked the idea and felt there were just enough options to make it viable, but figured it would do to see what everyone else thought.

**Comment by AurouraPlays** [comment]
> Poly refuses to go in the extraction portal. Extracting Renault did work, though. Round ID was 49881, if it helps.

Looking into this, verified that it was working as intended. Renault was the target, so the portal refused to take Poly. I tested to make sure, and Poly does go into the portal when she is the target.

**Comment by AurouraPlays** [comment]
> Poly refuses to go in the extraction portal. Extracting Renault did work, though. Round ID was 49881, if it helps.

On investigation, this turned out to be an issue with the checks in place that prevented multiple unique objectives from rolling simultaneously. When having two kidnap pet objectives, the checks on the flare for determining target don't work as intended.

This issue has been fixed in the form of fixing the duplicate-objective check.
